components/script/dom/htmlmediaelement.rs

```diff
@@ -1381,11 +1543,17 @@ impl VirtualMethods for HTMLMediaElement {
     fn attribute_mutated(&self, attr: &Attr, mutation: AttributeMutation) {
         self.super_type().unwrap().attribute_mutated(attr, mutation);
 
+        if mutation.new_value(attr).is_none() {
+            return;
+        }
+
         match attr.local_name() {
             &local_name!("src") => {
-                if mutation.new_value(attr).is_some() {
-                    self.media_element_load_algorithm();
-                }
+                self.media_element_load_algorithm();
+            },
+            &local_name!("poster") => {
+                let video_element = self.downcast::<HTMLVideoElement>().unwrap();
```

Comment:
We don't want to panic if we set a poster attribute on an audio element, and the easiest way to accomplish that is move this code to HTMLVideoElement instead.